python/lsst/ts/atspectrograph/atspec_csc.py

```diff
@@ -497,11 +485,24 @@ async def move_element(self, query, move, position, report, inposition,
 
             if (state[0] == ATSpectrograph.Status.STATIONARY and
                     state[1]-position <= self.model.tolerance):
-                if position_name is None:
+                if report == "reportedLinearStagePosition":
+                    # this is for the linear stage only since it's the only topic with a position attribute
                     getattr(self, f"evt_{report}").set_put(position=state[1])
-                else:
+                elif report == "reportedFilterPosition":
                     getattr(self, f"evt_{report}").set_put(position=state[1]+1,
-                                                           name=position_name)
+                                                           name=position_name,
+                                                           centralWavelength=
+                                                           self.filter_info['filter_central_wavelength'][state[1]],
+                                                           focusOffset=self.filter_info["filter_focus_offset"][
+                                                               int(state[1])] )
+                elif report == "reportedDisperserPosition":
+                    getattr(self, f"evt_{report}").set_put(position=state[1]+1,
+                                                           name=position_name,
+                                                           focusOffset=self.grating_info["grating_focus_offset"][
+                                                               int(state[1])])
+                else:
+                    raise RuntimeError(f"Expected report = reportedLinearStagePosition, reportedFilterPosition or "
```

Comment:
This is a good catch... I wonder if you should try to capture this error earlier... It would definitively make the code  more robust. You could add this at the start of the method:

```
if report not in ["reportedLinearStagePosition", "reportedFilterPosition", "reportedDisperserPosition"]:
    raise RuntimeError(f"Expected report = reportedLinearStagePosition, reportedFilterPosition or "
                                     f"reportedDisperserPosition, but got {report}")
```

then you if statement here could become

```
if report == "reportedFilterPosition":
    getattr(self, f"evt_{report}").set_put(position=state[1]+1,
                                           name=position_name,
                                           centralWavelength=
                                           self.filter_info['filter_central_wavelength'][state[1]],
                                           focusOffset=self.filter_info["filter_focus_offset"][
                                               int(state[1])] )
elif report == "reportedDisperserPosition":
    getattr(self, f"evt_{report}").set_put(position=state[1]+1,
                                           name=position_name,
                                           focusOffset=self.grating_info["grating_focus_offset"][
                                               int(state[1])])
else:
    getattr(self, f"evt_{report}").set_put(position=state[1])
```

Since linear stage is the least used...

Reply:
Done, but i left a comment saying the else was for the linear stage.... only for readability sake.